baselines/flanders/flanders/main.py

```diff
@@ -0,0 +1,277 @@
+        )
+    elif cfg.strategy.name == "bulyan":
+        # Get aggregation rule function
+        strategy = instantiate(
+            cfg.strategy.strategy,
+            evaluate_fn=clients[dataset_name][1],
+            on_fit_config_fn=fit_config,
+            fraction_fit=1,
+            fraction_evaluate=0,
+            min_fit_clients=cfg.server.pool_size,
+            min_evaluate_clients=0,
+            min_available_clients=cfg.server.pool_size,
+            num_malicious_clients=num_malicious,
+            to_keep=cfg.strategy.strategy.to_keep,
+        )
+    elif cfg.strategy.name == "trimmedmean":
+        strategy = instantiate(
+            cfg.strategy.strategy,
+            evaluate_fn=clients[dataset_name][1],
+            on_fit_config_fn=fit_config,
+            fraction_fit=1,
+            fraction_evaluate=0,
+            min_fit_clients=cfg.server.pool_size,
+            min_evaluate_clients=0,
+            min_available_clients=cfg.server.pool_size,
+            beta=cfg.strategy.strategy.beta,
+        )
+    elif cfg.strategy.name == "fedmedian":
+        strategy = instantiate(
+            cfg.strategy.strategy,
+            evaluate_fn=clients[dataset_name][1],
+            on_fit_config_fn=fit_config,
+            fraction_fit=1,
+            fraction_evaluate=0,
+            min_fit_clients=cfg.server.pool_size,
+            min_evaluate_clients=0,
+            min_available_clients=cfg.server.pool_size,
+        )
+    else:
+        raise ValueError("Strategy not supported")
+
+    # 5. Start Simulation
+    history = fl.simulation.start_simulation(
+        client_fn=client_fn,
+        num_clients=cfg.server.pool_size,
+        client_resources=cfg.client_resources,
+        server=EnhancedServer(
+            warmup_rounds=cfg.server.warmup_rounds,
+            num_malicious=num_malicious,
+            attack_fn=attacks[attack_fn],  # type: ignore
+            magnitude=cfg.server.magnitude,
+            client_manager=SimpleClientManager(),
+            strategy=strategy,
+            sampling=cfg.server.sampling,
+            history_dir=cfg.server.history_dir,
+            dataset_name=dataset_name,
+            threshold=cfg.server.threshold,
+            omniscent=cfg.server.omniscent,
+        ),
+        config=fl.server.ServerConfig(num_rounds=cfg.server.num_rounds),
+        strategy=strategy,
+    )
+
+    save_path = HydraConfig.get().runtime.output_dir
+
+    rounds, test_loss = zip(*history.losses_centralized)
+    _, test_accuracy = zip(*history.metrics_centralized["accuracy"])
+    _, test_auc = zip(*history.metrics_centralized["auc"])
+    _, truep = zip(*history.metrics_centralized["TP"])
+    _, truen = zip(*history.metrics_centralized["TN"])
+    _, falsep = zip(*history.metrics_centralized["FP"])
+    _, falsen = zip(*history.metrics_centralized["FN"])
+
+    path_to_save = [os.path.join(save_path, "results.csv"), "outputs/all_results.csv"]
```

Comment:
I think the save in `outputs/all_results.csv` is going to fail if no `outputs/` dir exist. This is the case actually if a person runs your `--multirun` commands directly (which don't create the `outputs/` dir).

Reply:
Right! Good point